console-subscriber/src/aggregator.rs

```diff
@@ -615,9 +906,287 @@ impl Task {
     }
 }
 
+impl ToProto for TaskStats {
+    type Result = proto::tasks::Stats;
+
+    fn to_proto(&self) -> Self::Result {
+        proto::tasks::Stats {
+            poll_stats: Some(self.poll_stats.to_proto()),
+            created_at: self.created_at.map(Into::into),
+            total_time: total_time(self.created_at.as_ref(), self.closed_at.as_ref())
+                .map(Into::into),
+            wakes: self.wakes,
+            waker_clones: self.waker_clones,
+            waker_drops: self.waker_drops,
+            last_wake: self.last_wake.map(Into::into),
+        }
+    }
+}
+
+impl ToProto for Resource {
+    type Result = proto::resources::Resource;
+
+    fn to_proto(&self) -> Self::Result {
+        proto::resources::Resource {
+            id: Some(self.id.clone().into()),
+            kind: Some(self.kind.clone()),
+            metadata: Some(self.metadata.into()),
+            concrete_type: self.concrete_type.clone(),
+        }
+    }
+}
+
+impl ToProto for ResourceStats {
+    type Result = proto::resources::Stats;
+
+    fn to_proto(&self) -> Self::Result {
+        let attributes = self
+            .attributes
+            .values()
+            .cloned()
```

Comment:
nit/tioli: i *think* we should be able to remove the `cloned()` here, since `to_proto` takes a reference:
```suggestion
```